src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -313,14 +497,249 @@ private void updateCitizens()
+        townHall.getColony().getPermissions().togglePermission(rank, action);
+
+        if (trigger)
+        {
+            button.setLabel(LanguageHandler.format(OFF));
+        }
+        else
+        {
+            button.setLabel(LanguageHandler.format(ON));
+        }
+    }
+
+    /**
+     * Switch between previous and next page.
+     */
+    private void switchPage(@NotNull final Button button)
+    {
+        if (button.getID().equals(BUTTON_PREV_PAGE_PERM))
+        {
+            findPaneOfTypeByID(VIEW_PERM_PAGES, SwitchView.class).previousView();
+
+            findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setVisible(false);
+            findPaneOfTypeByID(BUTTON_NEXT_PAGE_PERM, Button.class).setVisible(true);
+        }
+        else
+        {
+            findPaneOfTypeByID(VIEW_PERM_PAGES, SwitchView.class).nextView();
+
+            findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setVisible(true);
+            findPaneOfTypeByID(BUTTON_NEXT_PAGE_PERM, Button.class).setVisible(false);
+        }
+
+        if (findPaneOfTypeByID(VIEW_PERM_PAGES, SwitchView.class).getCurrentView().getID().equals(PERMISSION_VIEW))
+        {
+            findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setVisible(true);
+            findPaneOfTypeByID(BUTTON_NEXT_PAGE_PERM, Button.class).setVisible(true);
+
+            fillPermissionList(VIEW_OFFICER);
+        }
+    }
+
+    /**
+     * Fills the permission list in the GUI.
+     */
+    private void fillPermissionList(@NotNull final String category)
+    {
+        actionsList = findPaneOfTypeByID(LIST_ACTIONS + category, ScrollingList.class);
+        actionsList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return Action.values().length - IGNORE_INDEX;
+            }
+
+            @Override
+            public void updateElement(final int index, @NotNull final Pane rowPane)
+            {
+                final int actionIndex = index <= IGNORE_INDEX ? index : (index + IGNORE_INDEX);
+                final Action action = Action.values()[actionIndex];
+                final String name = LanguageHandler.format(KEY_TO_PERMISSIONS + action.toString().toLowerCase(Locale.US));
+
+                if (name.contains(KEY_TO_PERMISSIONS))
+                {
+                    Log.getLogger().warn("Didn't work for:" + name);
+                    return;
+                }
+
+                rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(name);
+                final boolean isTriggered = townHall.getColony().getPermissions().hasPermission(Rank.valueOf(actionsList.getParent().getID().toUpperCase(Locale.ENGLISH)), action);
+                rowPane.findPaneOfTypeByID("trigger", Button.class)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'BUTTON_TRIGGER' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)